### ISSUE TITLE
Der Swap wurde nicht zurück erstattet.

### ISSUE THREAD

**Issue by Alkarz** [DeFiCh/app#318]
Hi.
Am 02.01.21 08:50 am habe Liquidity USDT hinzufügt in die Summe 30,64051312 USDT und mir würde vom mein Wallet automatisch Swap abgezogen, in die Menge 19,94553149 DFI Coin. Zu diese Zeitpunkt am 2.01.21 ein DFI Coin kostete 1,53620941 USDT. Das heißt zusammen gerechnet habe ich 61,28102628 USDT für die Liquidity mining hinterlegt.

19,94553149 DFI*1,53620941 USDT=30,64051316 USDT + 30,64051312 USDT= 61,28102628 USDT

Am 15.01.21 habe ich Liquidity USDT auf meinen DeFi Chain entfernt und 40,74439752 USDT auf mein USDT-Wallet bekamm, aber Swap der mir am 02.01.21 von mein DFI-Wallet abgezogen wurde, habe ich nicht zurück erstattet bekamm. Die menge des verschwundenen Coin die ich beträgt 15.08932607 DFI, Das heißt für mich dass ich ein Verlustgeschäft gemacht habe, in die summe -20,53662876 USDT

Liquidity hinzugefügt 19,94553149 DFI + 30,64051312 USDT= - 61,28102628 USDT
Liquidity entfernt 0,00000000 DFI + 40,74439752 USDT
Verlust -20,53662876 USDT

Ich möchte gerne meine 15.08932607 DFI Coin zurück haben. Ich bitte Sie um die Klarstellung.
Mit freundlichen Grüßen Alkarz.

**Comment by izzycsy** [comment]
Hi @Alkarz , thanks for your feedback. Can I have the transaction id for the following transactions you mentioned? 


Could you also type the following commands via the CLI and send output: 

`getbalance` — utxo DFI
`gettokenbalances` — DFI 

 The sum of utxo DFI & DFI would be your total tokens.

`listaccounts {} true true true` — breakdown of your coin(s) / token(s)

**Comment by Alkarz** [comment]
Ich bin kein Programmierer.
Dies Ergebnis habe ich von Befehlskonsole bekommen konnte .
Ich habe keine Ahnung was ich damit machen soll.


//
getbalance
1.20494005
gettokenbalances
[
  "2.13303111@0",
  "0.00000652@1",
  "0.00000001@2",
  "91.95158035@3",
  "4.73506926@4",
  "0.37158776@5"
]
listaccounts {} true true true
[
  {
    "key": "01010101010101010101011",
    "owner": {
      "asm": "OP_HASH160 0000000000000000000000000000P_EQUAL",
      "hex": "000000000000000000000000000000000000000007",
      "reqSigs": 1,
      "type": "scripthash",
      "addresses": [
        "00000000000000000000000000000000000000"
      ]
    },
    "amount": {
      "0": 1.07509277
    }
  },
  {
    "key": "01010101010101010101011",
    "owner": {
      "asm": "OP_HASH160 101010101010101 OP_EQUAL",
      "hex": "01010101010101010101011",
      "reqSigs": 1,
      "type": "scripthash",
      "addresses": [
        "000000000000000000000000000000000000000V"
      ]
    },
    "amount": {
      "2": 1e-8
    }
  },
  {
    "key": "01010101010101010101011",
    "owner": {
      "asm": "OP_HASH160 0000000000000000000000000000000000000000000000000EQUAL",
      "hex": "01010101010101010101011",
      "reqSigs": 1,
      "type": "scripthash",
      "addresses": [
        "dEStXxpxjNgTZd8BDikmudL4x26tU4g5gV"
      ]
    },
    "amount": {
      "5": 0.37158776
    }
  },
  {
    "key": "010101010100000000000000000000000000000",
    "owner": {
      "asm": "OP_HASH160 0000000000000000000000000000000 OP_EQUAL",
      "hex": "a000000000000000000000000000000000000000000000000000",
      "reqSigs": 1,
      "type": "scripthash",
      "addresses": [
        "0000000000000000000000000000000000"
      ] 
    },
    "amount": {
      "0": 1.06645412
    }
  },
  {
    "key": "a00000000000000000000000000000000000000000000001",
    "owner": {
      "asm": "000000000000000000000000000000000000000000000000000AL",
      "hex": "00000000000000000000000000000000000000000000",
      "reqSigs": 1,
      "type": "scripthash",
      "addresses": [
        "00000000000000000000000000000000000000000"
      ]
    },
    "amount": {
      "1": 0.00000652
    }
  },
  {
    "key": "0000000000000000000000000000000000000000000000003",
    "owner": {
      "asm": "OP_HASH160 000000000000000000000000000000000UAL",
      "hex": "000000000000000000000000000000000000000000000",
      "reqSigs": 1,
      "type": "scripthash",
      "addresses": [
        "00000000000000000000000000000000000000000000000H"
      ]
    },
    "amount": {
      "3": 91.95158035
    }
  },
  {
    "key": "00000000000000000000000000000000000000000000000000000000000",
    "owner": {
      "asm": "00000000000000000000000000000000000000000000_EQUAL",
      "hex": "0000000000000000000000000000000000000000000000000000000007",
      "reqSigs": 1,
      "type": "scripthash",
      "addresses": [
        "000000000000000000000000000000000000000000H"
      ]
    },
    "amount": {
      "4": 4.73506926
    }
  }
]
//

**Comment by izzycsy** [comment]
Hi @Alkarz , i understand. Thanks for the output. Can I have the address for this? I would like to check on Mainnet <https://mainnet.defichain.io/#/DFI/mainnet/home> to further analyse your transactions. Thank you. 

```
{
"key": "0000000000000000000000000000000000000000000000003",
"owner": {
"asm": { "asm": { "asm": { "asm": { " "OP_HASH160 000000000000000000000000000000000UAL",
"hex" "000000000000000000000000000000000000000000000"
"reqSigs" 1,
"type": "scripthash",
"addresses": [
"00000000000000000000000000000000000000000000000H"
]
}
"amount":{
"3": 91.95158035
}
}
```

**Comment by Alkarz** [comment]
Hi. Danke dir für deine Unterstützung.
Hier ist noch mal Ergebnis mit Adress.

listaccounts {} true true true
[
  {
    "key": "00000000000000000000000000000000000000000000",
    "owner": {
      "asm": "000000000000000000000000000000000 ",
      "hex": "00000000000000000000000000000000000000000",
      "reqSigs": 1,
      "type": "scripthash",
      "addresses": [
        "dEStXxpxjNgTZd8BDikmudL4x26tU4g5gV"
      ]
    },
    "amount": {
      "0": 1.59120733
    }
  },
  {
    "key": "0000000000000000000000000000000000000000000000000",
    "owner": {
      "asm": "000000000000000000000000000000000000000",
      "hex": "a000000000000000000000000000000000000000000000",
      "reqSigs": 1,
      "type": "scripthash",
      "addresses": [
        "dEStXxpxjNgTZd8BDikmudL4x26tU4g5gV"
      ]
    },
    "amount": {
      "2": 1e-8
    }
  },
  {
    "key": "0000000000000000000000000000000000000000",
    "owner": {
      "asm": "00000000000000000000000000000000000000",
      "hex": "a0000000000000000000000000000000000000000000",
      "reqSigs": 1,
      "type": "scripthash",
      "addresses": [
        "dEStXxpxjNgTZd8BDikmudL4x26tU4g5gV"
      ]
    },
    "amount": {
      "5": 0.37158776
    }
  },
  {
    "key": "00000000000000000000000000000000000000000000000000000",
    "owner": {
      "asm": "00000000000000000000000000000000000000000000000000L",
      "hex": "000000000000000000000000000000000000000000000000000007",
      "reqSigs": 1,
      "type": "scripthash",
      "addresses": [
        "dW16tShWbTRv1kvDzCs9QrzmstdCTyxdjH"
      ]
    },
    "amount": {
      "0": 2.22315599
    }
  },
  {
    "key": "000000000000000000000000000000000000000001",
    "owner": {
      "asm": "OP_HASH16000000000000000000000000000000000000000L",
      "hex": "000000000000000000000000000000000000000007",
      "reqSigs": 1,
      "type": "scripthash",
      "addresses": [
        "dW16tShWbTRv1kvDzCs9QrzmstdCTyxdjH"
      ]
    },
    "amount": {
      "1": 0.00001229
    }
  },
  {
    "key": "a000000000000000000000000000000000000000000000003",
    "owner": {
      "asm": "OP_0000000000000000000000000000000000000000000000000QUAL",
      "hex": "00000000000000000000000000000000000000000000",
      "reqSigs": 1,
      "type": "scripthash",
      "addresses": [
        "dW16tShWbTRv1kvDzCs9QrzmstdCTyxdjH"
      ]
    },
    "amount": {
      "3": 91.95158035
    }
  },
  {
    "key": "00000000000000000000000000000000000000",
    "owner": {
      "asm": "00000000000000000000000000000000000000000000000",
      "hex": "0000000000000000000000000000000000000000000000",
      "reqSigs": 1,
      "type": "scripthash",
      "addresses": [
        "dW16tShWbTRv1kvDzCs9QrzmstdCTyxdjH"
      ]
    },
    "amount": {
      "4": 4.73506926
    }
  }
]


getbalance
1.20494005

gettokenbalances
[
  "3.82202076@0",
  "0.00001229@1",
  "0.00000001@2",
  "91.95158035@3",
  "4.73506926@4",
  "0.37158776@5"

**Comment by Alkarz** [comment]
"addresses": [
"dEStXxpxjNgTZd8BDikmudL4x26tU4g5gV"

**Comment by izzycsy** [comment]
Hi @Alkarz , I understand now, I would like you to help you better understand how Liquidity Mining works. As your thought process of calculation is incorrect. Using your scenario: 

When you added `30.64051312 USDT` + `19.94553149` to Liquidity Pool, it equals to `XX USDT-DFI`. The two tokens become a pair named **USDT-DFI**, and would be the amount you started in the Liquidity Pool.

```
19.94553149 DFI * 1.53620941 USDT = 30.64051316 USDT
```
Only is you use DEX to swap DFI **to** USDT. 


Now, back to your case. There is possibly 2 scenarios:
1. You did actually received the `15.08932607`, just that it's reflected in the `USDT-DFI` wallet transaction history but not reflected in `DFI` wallet's transaction history. Meaning your balances are correct

_or_ 

2. You did not receive the `15.08932607`. 

We've to investigate further. Can you try to `Refresh UTXO` in Settings, and see if the `15.08932607` is reflected in your DFI wallet transaction history. 

<img width="777" alt="Refresh UTXOs" src="https://user-images.githubusercontent.com/65014479/105195578-bae80080-5b75-11eb-9d77-c85d74af4d9e.png">

**Comment by izzycsy** [comment]
Hi @Alkarz , if the transaction `15.08932607 DFI` still does not appear in your wallet, please send your app logs. 

<img width="339" alt="1 App log" src="https://user-images.githubusercontent.com/65014479/105198214-64c88c80-5b78-11eb-869d-0f6cb4de8990.png">

<img width="511" alt="2 Komprimiere" src="https://user-images.githubusercontent.com/65014479/105198226-6a25d700-5b78-11eb-9b35-42a2ea048126.png">

**Comment by Alkarz** [comment]
Hi izzycsy.
[binary_log-1.zip](https://github.com/DeFiCh/app/files/5852225/binary_log-1.zip)

https://user-images.githubusercontent.com/77533825/105413535-8425ef00-5c36-11eb-8d5f-83febdaec032.mp4

**Comment by izzycsy** [comment]
Hi @Alkarz, thanks for the screen record. Please give us time to investigate the possibly missing 15 DFI. Next, regarding the DEX issue you're facing, a fix will be included in the next app update. Here's a link to the issue in the mean time. <https://github.com/DeFiCh/app/issues/280>

**Comment by Alkarz** [comment]
Hi Bella izzycsy , wie sieht aus. Haben Sie was rausgefunden wo meine 15 DFI stecken geblieben sind? Vielen Dank in voraus.

**Comment by izzycsy** [comment]
Hi @Alkarz , we need to trace your transactions so it's taking awhile, I'll update you once the investigation is complete. Thank you for your patience.

**Comment by izzycsy** [comment]
Hi @Alkarz , thank you for your patience. We did some investigation, and this is what we found: 

at block `# 574646`, your balance before you did anything was `0.01136818@DFI`.
blockHash: `c281189a561cc170a7ae9e4f6d5c856488df5df207bccdd99b1d884b3475883c`

Then you proceeded to:
```
RemovePoolLiquidity
"amounts": [
            "15.08932607@DFI",
            "40.74439752@USDT",
            "-24.79692011@USDT-DFI"
        ]
```

The rewards from RemovePoolLiquidity was successful in block `# 574647`
blockHash: `f0fc5d66b1f1bb7b9223f829ebdba62a4701d6ca6d82c58b626385353e89e6a6`


In the next few blocks you did the following: 
block `# 574651`
```
AccountToAccount 
"amounts": [
            “0.00593624@DFI"
        ]
```

block `# 574653`
```
UtxosToAccount
"amounts": [
            "3.89056171@DFI"
        ]
```

block `# 574654`
```
poolSwap
"amounts": [
            "-19.00000000@DFI",
            "51.20009479@USDT"
        ]
```

At this point your DFI balance was more then the sum of DFI in block ` # 574646` , `# 574647` , `# 574651` and `# 574653` combined, hence you were able to poolSwap `19.00000000@DFI` to  `51.20009479@USDT`.  

In conclusion, your DFI was sent to your account there's no missing `15.08932607 DFI`. Please do understand that DeFiChain an open source project and that your funds are your own responsibility, what we can do is try to help investigate where the funds went. Transactions are either processed or not processed, there's no "halfway" state where it gets stuck in the middle. Thank you for your understanding and patience once again.
